csharp/src/Apache.Arrow/Memory/ILargeNativeAllocationOwner.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Apache.Arrow.Memory
+{
+    internal interface ILargeNativeAllocationOwner
```

Comment:
I don't think we need a separate interface here, we can just add a new overload of the `Release` method to `INativeAllocationOwner`.